contrib/generate_precompile.jl

```diff
@@ -183,10 +183,11 @@ for match = Base._methods(+, (Int, Int), -1, Base.get_world_counter())
     # interactive startup uses this
     write(IOBuffer(), "")
 
-    # not critical, but helps hide unrelated compilation from @time when using --trace-compile
-    foo() = rand(2,2) * rand(2,2)
-    @time foo()
-    @time foo()
+    # Not critical, but helps hide unrelated compilation from @time when using --trace-compile.
+    # Do not call `rand` which can cause bad caching on some platforms: #56177.
+    f55729() = ones(2, 2) * ones(2, 2)
+    @time f55729()
+    @time f55729()
```

Comment:
The purpose of this was to hit allocation & compilation so that the macro printing is fully exercised. Perhaps this is a more robust way to do that, using `Experimental.@force_compile` and new lightweight functions.

```suggestion
    # Not critical, but helps hide unrelated compilation from @time when using --trace-compile.
    # Do not call `rand` which can cause bad caching on some platforms: #56177.
    f55729() = Base.Experimental.@force_compile
    @time @eval f55729()
    @time @eval f55729()
```
i.e.
```
julia> @time @eval f55729()
precompile(Tuple{Type{NamedTuple{(:value, :time, :bytes, :gctime, :gcstats, :lock_conflicts, :compile_time, :recompile_time), T} where T<:Tuple}, Tuple{Nothing, Float64, Int64, Float64, Base.GC_Diff, Int64, Float64, Float64}})
precompile(Tuple{typeof(Base.getproperty), NamedTuple{(:value, :time, :bytes, :gctime, :gcstats, :lock_conflicts, :compile_time, :recompile_time), Tuple{Nothing, Float64, Int64, Float64, Base.GC_Diff, Int64, Float64, Float64}}, Symbol})
precompile(Tuple{typeof(Core.kwcall), NamedTuple{(:msg,), Tuple{Nothing}}, typeof(Base.time_print), Base.TTY, Float64, Int64, Int64, Int64, Int64, Float64, Float64, Bool})
  0.000106 seconds (144 allocations: 6.594 KiB, 19.49% compilation time)
```

Reply:
I checked that'd still work, thanks.